tv_apps/browser/js/bookmarkStore.js

```diff
@@ -99,33 +141,50 @@
             // return Promise.
             SyncBrowserDB.getBookmark({
               parentid: SYNC_BOOKMARK_ROOT_FOLDER_ID
-            }, syncBookmark => {
+            }, syncPlacesBookmark => {
               // We query remote bookmarks by parentid with
               // SYNC_BOOKMARK_ROOT_FOLDER_ID for checking if the indexedDB has
               // remote bookmark data. If there is any remote data, we create a
               // folder item to show the synced bookmark.
-              if(syncBookmark.length > 0) {
-                this.cache.push({
-                  id: SYNC_BOOKMARK_ROOT_FOLDER_ID,
-                  // XXX we need UX's suggestion to have a folder name for root
-                  // folder and don't forget to handle l10n case.
-                  title: 'Synced Bookmarks',
-                  type: 'folder',
-                  readOnly: true
+              if(syncPlacesBookmark.length > 0) {
```

Comment:
If 'mobile' folder is not in syncPlacesBookmark, it is still possible that there is a 'mobile' folder record in SyncBrowserDB. Please consider the case for mobile folder separately.